runner/src/server/plugins/engine/models/submission/WebhookModel.ts

```diff
@@ -2,6 +2,22 @@ import { DetailItem } from "../types";
 import { format } from "date-fns";
 import config from "server/config";
 
+function flattenMultiInput(rawValue) {
+  if (!rawValue) return rawValue;
+  rawValue.forEach((value) => {
+    Object.keys(value).map((key) => {
+      if (typeof value[key] == "object" && value[key] !== null) {
+        value[key] = Object.values(value[key])
+          .filter((p) => {
+            return !!p;
+          })
+          .join(", ");
+      }
+    });
+  });
+  return rawValue;
+}
```

Comment:
Shouldn't we sort of do this the other way round? Instead of changing the structured object to a string joined by commas, change the other way so that it matches the structured object?  

Just in the past seen dates stored as raw strings and then end up getting a requirement or request to re-display specific parts, which ends up in using some library (libpostal) to parse the address or some hacky regex to extract specific parts (i.e postcode) - it'd be better if we store the structured form of the address and then just on any relevant view display it as a comma separated string, or any other display that is requested

Reply:
Most of the code on the assess side for rendering is already built for formatting the string that this creates and they are having errors on their side of the json object appearing in the tables so this fixes both of these issues though it may be good to have structure in the future by removing this function

Reply:
Cool - how are we handling the state already stored in the database for this component? Do we need a migration script to replicate this change for persisted data with the old structure?